cmake/common.cmake

```diff
@@ -20,7 +20,7 @@ function(target_warnings_as_errors _TARGET)
   if(${_INDEX} GREATER -1)
     set(_COMPILER_WAE -Wall -W -Werror)
   elseif("${CMAKE_CXX_COMPILER_ID}" STREQUAL "MSVC")
-    set(_COMPILER_WAE /W4 /WX)
```

Comment:
Why are we turning off `/W4` ? seems like a less than ideal thing to be doing...

Reply:
Because in my setup, I got warnings from MSVC about us having added `/W4` when `/W3` was already set. IOW, `/W3` is what CMake uses for MSVC by default. `/W3` also happens to be what MSVC recommends for production code. Finally, `/W4` will inevitably be more pedantic and, truth be told, I would rather rely on gcc and clang to provide me with useful/meaningful warnings (YMMV).

Reply:
> I would rather rely on gcc and clang to provide me with useful/meaningful warnings (YMMV).

Thats fine for those developing on those platforms, but given we had `/W4` and it is working outside of this PR, what is it in this change that alters that and do we need it? From my brief read, this would be equivalent to turning off `-Wall` for other compilers, and I'd guess that would be a big issue.

Reply:
Nothing in this PR forced me to replace `/W4` with `/W3`. As it happens, I have the same problem when trying to build libCellML using the `develop` branch (using Qt Creator, which is what I also use on macOS):

```
13:36:43: Running steps for project libCellML...
13:36:43: Persisting CMake state...
13:36:52: Starting: "C:\Program Files\CMake\bin\cmake.exe" --build . --target all
[1/168 0.1/sec] Building CXX object src\CMakeFiles\cellml.dir\importedentity.cpp.obj
cl : Command line warning D9025 : overriding '/W3' with '/W4'
[2/168 0.3/sec] Building CXX object src\CMakeFiles\cellml.dir\entity.cpp.obj
cl : Command line warning D9025 : overriding '/W3' with '/W4'
[3/168 0.4/sec] Building CXX object src\CMakeFiles\cellml.dir\logger.cpp.obj
cl : Command line warning D9025 : overriding '/W3' with '/W4'
[4/168 0.4/sec] Building CXX object src\CMakeFiles\cellml.dir\issue.cpp.obj
cl : Command line warning D9025 : overriding '/W3' with '/W4'
[5/168 0.5/sec] Building CXX object src\CMakeFiles\cellml.dir\generatorprofile.cpp.obj
cl : Command line warning D9025 : overriding '/W3' with '/W4'
[6/168 0.5/sec] Building CXX object src\CMakeFiles\cellml.dir\importsource.cpp.obj
cl : Command line warning D9025 : overriding '/W3' with '/W4'
[7/168 0.4/sec] Building CXX object src\CMakeFiles\cellml.dir\generator.cpp.obj
cl : Command line warning D9025 : overriding '/W3' with '/W4'
[8/168 0.5/sec] Building CXX object src\CMakeFiles\cellml.dir\parser.cpp.obj
cl : Command line warning D9025 : overriding '/W3' with '/W4'
[9/168 0.5/sec] Building CXX object src\CMakeFiles\cellml.dir\namedentity.cpp.obj
cl : Command line warning D9025 : overriding '/W3' with '/W4'
[10/168 0.6/sec] Building CXX object src\CMakeFiles\cellml.dir\printer.cpp.obj
cl : Command line warning D9025 : overriding '/W3' with '/W4'
[11/168 0.5/sec] Building CXX object src\CMakeFiles\cellml.dir\reset.cpp.obj
cl : Command line warning D9025 : overriding '/W3' with '/W4'
[12/168 0.5/sec] Building CXX object src\CMakeFiles\cellml.dir\model.cpp.obj
cl : Command line warning D9025 : overriding '/W3' with '/W4'
[13/168 0.6/sec] Building CXX object src\CMakeFiles\cellml.dir\xmlattribute.cpp.obj
cl : Command line warning D9025 : overriding '/W3' with '/W4'
[14/168 0.6/sec] Building CXX object src\CMakeFiles\cellml.dir\units.cpp.obj
cl : Command line warning D9025 : overriding '/W3' with '/W4'
[15/168 0.6/sec] Building CXX object src\CMakeFiles\cellml.dir\variable.cpp.obj
cl : Command line warning D9025 : overriding '/W3' with '/W4'
[16/168 0.6/sec] Building CXX object src\CMakeFiles\cellml.dir\utilities.cpp.obj
cl : Command line warning D9025 : overriding '/W3' with '/W4'
[17/168 0.6/sec] Building CXX object src\CMakeFiles\cellml.dir\version.cpp.obj
cl : Command line warning D9025 : overriding '/W3' with '/W4'
[18/168 0.7/sec] Building CXX object src\CMakeFiles\cellml.dir\xmldoc.cpp.obj
cl : Command line warning D9025 : overriding '/W3' with '/W4'
[19/168 0.7/sec] Building CXX object src\CMakeFiles\cellml_debug_utilities.dir\debug.cpp.obj
[20/168 0.7/sec] Linking CXX static library src\cellml_debug_utilities.lib
[21/168 0.7/sec] Building CXX object src\CMakeFiles\cellml.dir\xmlutils.cpp.obj
cl : Command line warning D9025 : overriding '/W3' with '/W4'
[22/168 0.7/sec] Building CXX object src\CMakeFiles\cellml.dir\validator.cpp.obj
cl : Command line warning D9025 : overriding '/W3' with '/W4'
[23/168 0.7/sec] Building CXX object src\CMakeFiles\cellml.dir\xmlnode.cpp.obj
cl : Command line warning D9025 : overriding '/W3' with '/W4'
[24/168 0.8/sec] Building CXX object src\CMakeFiles\cellml.dir\componententity.cpp.obj
cl : Command line warning D9025 : overriding '/W3' with '/W4'
[25/168 0.8/sec] Building CXX object tests\CMakeFiles\test_utils.dir\test_utils.cpp.obj
[26/168 0.8/sec] Building CXX object tests\CMakeFiles\test_api_headers_generator.dir\api_headers\api_header_generator.cpp.obj
[27/168 0.8/sec] Building CXX object src\CMakeFiles\cellml.dir\component.cpp.obj
cl : Command line warning D9025 : overriding '/W3' with '/W4'
[28/168 0.8/sec] Building CXX object tests\CMakeFiles\test_resolve_imports.dir\resolve_imports\file_parser.cpp.obj
cl : Command line warning D9025 : overriding '/W3' with '/W4'
[29/168 0.8/sec] Linking CXX shared library src\libcellmld.dll
   Creating library src\libcellmld.lib and object src\libcellmld.exp
   Creating library src\libcellmld.lib and object src\libcellmld.exp
[30/168 0.8/sec] Building CXX object tests\CMakeFiles\test_component.dir\component\encapsulation.cpp.obj
cl : Command line warning D9025 : overriding '/W3' with '/W4'
[31/168 0.8/sec] Building CXX object tests\CMakeFiles\test_component.dir\component\component.cpp.obj
cl : Command line warning D9025 : overriding '/W3' with '/W4'
[32/168 0.8/sec] Swig compile ../interface/printer.i for python
[33/168 0.8/sec] Swig compile ../interface/reset.i for python
[34/168 0.8/sec] Swig compile ../interface/generatorprofile.i for python
[35/168 0.8/sec] Building CXX object tests\CMakeFiles\test_clone.dir\clone\clone.cpp.obj
cl : Command line warning D9025 : overriding '/W3' with '/W4'
[36/168 0.9/sec] Building CXX object src\bindings\python\CMakeFiles\printer.dir\swig_intermediaries\printerPYTHON_wrap.cxx.obj
[37/168 0.9/sec] Building CXX object src\bindings\python\CMakeFiles\reset.dir\swig_intermediaries\resetPYTHON_wrap.cxx.obj
[38/168 0.9/sec] Building CXX object src\bindings\python\CMakeFiles\generatorprofile.dir\swig_intermediaries\generatorprofilePYTHON_wrap.cxx.obj
[39/168 0.9/sec] Swig compile ../interface/component.i for python
[40/168 0.9/sec] Building CXX object tests\CMakeFiles\test_generator.dir\generator\generator.cpp.obj
cl : Command line warning D9025 : overriding '/W3' with '/W4'
[41/168 0.9/sec] Building CXX object tests\CMakeFiles\test_isolated_generator.dir\isolated\generator.cpp.obj
cl : Command line warning D9025 : overriding '/W3' with '/W4'
[42/168 1.0/sec] Linking CXX shared module src\bindings\python\libcellml\_printer.pyd
   Creating library src\bindings\python\printer.lib and object src\bindings\python\printer.exp
   Creating library src\bindings\python\printer.lib and object src\bindings\python\printer.exp
[43/168 1.0/sec] Linking CXX shared module src\bindings\python\libcellml\_reset.pyd
   Creating library src\bindings\python\reset.lib and object src\bindings\python\reset.exp
   Creating library src\bindings\python\reset.lib and object src\bindings\python\reset.exp
[44/168 1.0/sec] Linking CXX shared module src\bindings\python\libcellml\_generatorprofile.pyd
   Creating library src\bindings\python\generatorprofile.lib and object src\bindings\python\generatorprofile.exp
   Creating library src\bindings\python\generatorprofile.lib and object src\bindings\python\generatorprofile.exp
[45/168 1.0/sec] Swig compile ../interface/entity.i for python
[46/168 1.0/sec] Swig compile ../interface/componententity.i for python
[47/168 1.0/sec] Swig compile ../interface/generator.i for python
[48/168 1.1/sec] Building CXX object src\bindings\python\CMakeFiles\component.dir\swig_intermediaries\componentPYTHON_wrap.cxx.obj
[49/168 1.1/sec] Swig compile ../interface/model.i for python
[50/168 1.1/sec] Swig compile ../interface/importedentity.i for python
[51/168 1.1/sec] Building CXX object src\bindings\python\CMakeFiles\entity.dir\swig_intermediaries\entityPYTHON_wrap.cxx.obj
[52/168 1.1/sec] Building CXX object src\bindings\python\CMakeFiles\componententity.dir\swig_intermediaries\componententityPYTHON_wrap.cxx.obj
[53/168 1.1/sec] Building CXX object src\bindings\python\CMakeFiles\generator.dir\swig_intermediaries\generatorPYTHON_wrap.cxx.obj
[54/168 1.2/sec] Building CXX object src\bindings\python\CMakeFiles\model.dir\swig_intermediaries\modelPYTHON_wrap.cxx.obj
[55/168 1.2/sec] Linking CXX shared module src\bindings\python\libcellml\_component.pyd
   Creating library src\bindings\python\component.lib and object src\bindings\python\component.exp
   Creating library src\bindings\python\component.lib and object src\bindings\python\component.exp
[56/168 1.2/sec] Building CXX object src\bindings\python\CMakeFiles\importedentity.dir\swig_intermediaries\importedentityPYTHON_wrap.cxx.obj
[57/168 1.2/sec] Linking CXX shared module src\bindings\python\libcellml\_componententity.pyd
   Creating library src\bindings\python\componententity.lib and object src\bindings\python\componententity.exp
   Creating library src\bindings\python\componententity.lib and object src\bindings\python\componententity.exp
[58/168 1.2/sec] Linking CXX shared module src\bindings\python\libcellml\_entity.pyd
   Creating library src\bindings\python\entity.lib and object src\bindings\python\entity.exp
   Creating library src\bindings\python\entity.lib and object src\bindings\python\entity.exp
[59/168 1.2/sec] Linking CXX shared module src\bindings\python\libcellml\_generator.pyd
   Creating library src\bindings\python\generator.lib and object src\bindings\python\generator.exp
   Creating library src\bindings\python\generator.lib and object src\bindings\python\generator.exp
[60/168 1.3/sec] Swig compile ../interface/issue.i for python
[61/168 1.3/sec] Linking CXX shared module src\bindings\python\libcellml\_model.pyd
   Creating library src\bindings\python\model.lib and object src\bindings\python\model.exp
   Creating library src\bindings\python\model.lib and object src\bindings\python\model.exp
[62/168 1.3/sec] Linking CXX shared module src\bindings\python\libcellml\_importedentity.pyd
   Creating library src\bindings\python\importedentity.lib and object src\bindings\python\importedentity.exp
   Creating library src\bindings\python\importedentity.lib and object src\bindings\python\importedentity.exp
[63/168 1.3/sec] Swig compile ../interface/importsource.i for python
[64/168 1.3/sec] Swig compile ../interface/variable.i for python
[65/168 1.3/sec] Swig compile ../interface/namedentity.i for python
[66/168 1.3/sec] Swig compile ../interface/version.i for python
[67/168 1.4/sec] Swig compile ../interface/validator.i for python
[68/168 1.4/sec] Building CXX object src\bindings\python\CMakeFiles\issue.dir\swig_intermediaries\issuePYTHON_wrap.cxx.obj
[69/168 1.4/sec] Building CXX object src\bindings\python\CMakeFiles\namedentity.dir\swig_intermediaries\namedentityPYTHON_wrap.cxx.obj
[70/168 1.4/sec] Building CXX object src\bindings\python\CMakeFiles\importsource.dir\swig_intermediaries\importsourcePYTHON_wrap.cxx.obj
[71/168 1.4/sec] Building CXX object src\bindings\python\CMakeFiles\variable.dir\swig_intermediaries\variablePYTHON_wrap.cxx.obj
[72/168 1.4/sec] Building CXX object src\bindings\python\CMakeFiles\version.dir\swig_intermediaries\versionPYTHON_wrap.cxx.obj
[73/168 1.4/sec] Linking CXX shared module src\bindings\python\libcellml\_issue.pyd
   Creating library src\bindings\python\issue.lib and object src\bindings\python\issue.exp
   Creating library src\bindings\python\issue.lib and object src\bindings\python\issue.exp
[74/168 1.4/sec] Linking CXX shared module src\bindings\python\libcellml\_importsource.pyd
   Creating library src\bindings\python\importsource.lib and object src\bindings\python\importsource.exp
   Creating library src\bindings\python\importsource.lib and object src\bindings\python\importsource.exp
[75/168 1.5/sec] Linking CXX shared module src\bindings\python\libcellml\_namedentity.pyd
   Creating library src\bindings\python\namedentity.lib and object src\bindings\python\namedentity.exp
   Creating library src\bindings\python\namedentity.lib and object src\bindings\python\namedentity.exp
[76/168 1.5/sec] Linking CXX shared module src\bindings\python\libcellml\_variable.pyd
   Creating library src\bindings\python\variable.lib and object src\bindings\python\variable.exp
   Creating library src\bindings\python\variable.lib and object src\bindings\python\variable.exp
[77/168 1.5/sec] Building CXX object src\bindings\python\CMakeFiles\validator.dir\swig_intermediaries\validatorPYTHON_wrap.cxx.obj
[78/168 1.5/sec] Linking CXX shared module src\bindings\python\libcellml\_version.pyd
   Creating library src\bindings\python\version.lib and object src\bindings\python\version.exp
   Creating library src\bindings\python\version.lib and object src\bindings\python\version.exp
[79/168 1.5/sec] Swig compile ../interface/logger.i for python
[80/168 1.5/sec] Swig compile ../interface/parser.i for python
[81/168 1.5/sec] Swig compile ../interface/units.i for python
[82/168 1.5/sec] Linking CXX shared module src\bindings\python\libcellml\_validator.pyd
   Creating library src\bindings\python\validator.lib and object src\bindings\python\validator.exp
   Creating library src\bindings\python\validator.lib and object src\bindings\python\validator.exp
[83/168 1.5/sec] Building CXX object src\bindings\python\CMakeFiles\logger.dir\swig_intermediaries\loggerPYTHON_wrap.cxx.obj
[84/168 1.5/sec] Building CXX object src\bindings\python\CMakeFiles\parser.dir\swig_intermediaries\parserPYTHON_wrap.cxx.obj
[85/168 1.5/sec] Building CXX object src\bindings\python\CMakeFiles\units.dir\swig_intermediaries\unitsPYTHON_wrap.cxx.obj
[86/168 1.5/sec] Linking CXX shared module src\bindings\python\libcellml\_logger.pyd
   Creating library src\bindings\python\logger.lib and object src\bindings\python\logger.exp
   Creating library src\bindings\python\logger.lib and object src\bindings\python\logger.exp
[87/168 1.5/sec] Linking CXX shared module src\bindings\python\libcellml\_parser.pyd
   Creating library src\bindings\python\parser.lib and object src\bindings\python\parser.exp
   Creating library src\bindings\python\parser.lib and object src\bindings\python\parser.exp
[88/168 1.5/sec] Linking CXX shared module src\bindings\python\libcellml\_units.pyd
   Creating library src\bindings\python\units.lib and object src\bindings\python\units.exp
   Creating library src\bindings\python\units.lib and object src\bindings\python\units.exp
[89/168 1.5/sec] Running utility command for python_bindings
[90/168 1.5/sec] Building CXX object tests\CMakeFiles\test_issue.dir\issue\issue.cpp.obj
cl : Command line warning D9025 : overriding '/W3' with '/W4'
[91/168 1.5/sec] Building CXX object tests\CMakeFiles\test_generator.dir\generator\generatorprofile.cpp.obj
cl : Command line warning D9025 : overriding '/W3' with '/W4'
[92/168 1.5/sec] Building CXX object tests\CMakeFiles\test_api_headers_logger.dir\api_headers\api_header_logger.cpp.obj
[93/168 1.5/sec] Building CXX object tests\CMakeFiles\test_connection.dir\connection\connection.cpp.obj
cl : Command line warning D9025 : overriding '/W3' with '/W4'
[94/168 1.5/sec] Building CXX object tests\CMakeFiles\test_api_headers_entity.dir\api_headers\api_header_entity.cpp.obj
[95/168 1.5/sec] Building CXX object tests\CMakeFiles\test_api_headers_generatorprofile.dir\api_headers\api_header_generatorprofile.cpp.obj
[96/168 1.5/sec] Building CXX object tests\CMakeFiles\test_math.dir\math\math.cpp.obj
cl : Command line warning D9025 : overriding '/W3' with '/W4'
[97/168 1.5/sec] Building CXX object tests\CMakeFiles\test_logger.dir\logger\logger.cpp.obj
cl : Command line warning D9025 : overriding '/W3' with '/W4'
[98/168 1.5/sec] Building CXX object tests\CMakeFiles\test_coverage.dir\coverage\coverage.cpp.obj
cl : Command line warning D9025 : overriding '/W3' with '/W4'
[99/168 1.4/sec] Building CXX object tests\CMakeFiles\test_api_headers_namedentity.dir\api_headers\api_header_namedentity.cpp.obj
[100/168 1.4/sec] Building CXX object tests\CMakeFiles\test_printer.dir\printer\printer.cpp.obj
cl : Command line warning D9025 : overriding '/W3' with '/W4'
[101/168 1.4/sec] Building CXX object tests\CMakeFiles\test_version.dir\version\version.cpp.obj
cl : Command line warning D9025 : overriding '/W3' with '/W4'
[102/168 1.4/sec] Building CXX object tests\CMakeFiles\test_api_headers_component.dir\api_headers\api_header_component.cpp.obj
[103/168 1.4/sec] Building CXX object tests\CMakeFiles\test_reset.dir\reset\reset.cpp.obj
cl : Command line warning D9025 : overriding '/W3' with '/W4'
[104/168 1.4/sec] Building CXX object tests\CMakeFiles\test_api_headers_componententity.dir\api_headers\api_header_componententity.cpp.obj
[105/168 1.4/sec] Building CXX object tests\CMakeFiles\test_units.dir\units\units.cpp.obj
cl : Command line warning D9025 : overriding '/W3' with '/W4'
[106/168 1.4/sec] Building CXX object tests\CMakeFiles\test_validator.dir\validator\validator.cpp.obj
cl : Command line warning D9025 : overriding '/W3' with '/W4'
[107/168 1.4/sec] Building CXX object tests\CMakeFiles\test_variable.dir\variable\variable.cpp.obj
cl : Command line warning D9025 : overriding '/W3' with '/W4'
[108/168 1.4/sec] Building CXX object tests\CMakeFiles\test_model.dir\model\model_flattening.cpp.obj
cl : Command line warning D9025 : overriding '/W3' with '/W4'
[109/168 1.4/sec] Building CXX object tests\CMakeFiles\test_model.dir\model\component_import.cpp.obj
cl : Command line warning D9025 : overriding '/W3' with '/W4'
[110/168 1.3/sec] Building CXX object tests\CMakeFiles\test_api_headers_variable.dir\api_headers\api_header_variable.cpp.obj
[111/168 1.3/sec] Building CXX object tests\CMakeFiles\test_model.dir\model\model.cpp.obj
cl : Command line warning D9025 : overriding '/W3' with '/W4'
[112/168 1.3/sec] Building CXX object tests\CMakeFiles\test_model.dir\model\units_import.cpp.obj
cl : Command line warning D9025 : overriding '/W3' with '/W4'
[113/168 1.3/sec] Building CXX object tests\CMakeFiles\test_api_headers_version.dir\api_headers\api_header_version.cpp.obj
[114/168 1.3/sec] Building CXX object tests\CMakeFiles\test_api_headers_importedentity.dir\api_headers\api_header_importedentity.cpp.obj
[115/168 1.4/sec] Building CXX object tests\CMakeFiles\test_parser.dir\parser\libxml_user.cpp.obj
cl : Command line warning D9025 : overriding '/W3' with '/W4'
[116/168 1.4/sec] Building CXX object tests\CMakeFiles\test_parser.dir\parser\parser.cpp.obj
cl : Command line warning D9025 : overriding '/W3' with '/W4'
[117/168 1.4/sec] Building CXX object tests\CMakeFiles\test_parser.dir\parser\file_parser.cpp.obj
cl : Command line warning D9025 : overriding '/W3' with '/W4'
[118/168 1.4/sec] Building CXX object tests\CMakeFiles\test_api_headers_reset.dir\api_headers\api_header_reset.cpp.obj
[119/168 1.4/sec] Building CXX object tests\CMakeFiles\test_api_headers_importsource.dir\api_headers\api_header_importsource.cpp.obj
[120/168 1.4/sec] Building CXX object tests\CMakeFiles\test_api_headers_issue.dir\api_headers\api_header_issue.cpp.obj
[121/168 1.4/sec] Building CXX object tests\CMakeFiles\test_api_headers_parser.dir\api_headers\api_header_parser.cpp.obj
[122/168 1.4/sec] Building CXX object tests\CMakeFiles\test_api_headers_model.dir\api_headers\api_header_model.cpp.obj
[123/168 1.4/sec] Building CXX object tests\CMakeFiles\test_api_headers_units.dir\api_headers\api_header_units.cpp.obj
[124/168 1.4/sec] Building CXX object tests\CMakeFiles\test_api_headers_printer.dir\api_headers\api_header_printer.cpp.obj
[125/168 1.4/sec] Building CXX object tests\CMakeFiles\test_api_headers_types.dir\api_headers\api_header_types.cpp.obj
[126/168 1.4/sec] Building CXX object tests\CMakeFiles\test_api_headers_validator.dir\api_headers\api_header_validator.cpp.obj
[127/168 1.4/sec] Building CXX object tests\gtest\CMakeFiles\gtest_main.dir\src\gtest_main.cc.obj
[128/168 1.4/sec] Building CXX object tests\gtest\CMakeFiles\gtest.dir\src\gtest-all.cc.obj
[129/168 1.4/sec] Linking CXX shared library tests\gtest\gtest.dll
   Creating library tests\gtest\gtest.lib and object tests\gtest\gtest.exp
   Creating library tests\gtest\gtest.lib and object tests\gtest\gtest.exp
[130/168 1.4/sec] Linking CXX shared library tests\test_utils.dll
   Creating library tests\test_utils.lib and object tests\test_utils.exp
   Creating library tests\test_utils.lib and object tests\test_utils.exp
[131/168 1.4/sec] Linking CXX shared library tests\gtest\gtest_main.dll
   Creating library tests\gtest\gtest_main.lib and object tests\gtest\gtest_main.exp
   Creating library tests\gtest\gtest_main.lib and object tests\gtest\gtest_main.exp
[132/168 1.4/sec] Linking CXX executable tests\test_api_headers_generator.exe
[133/168 1.4/sec] Linking CXX executable tests\test_issue.exe
[134/168 1.5/sec] Linking CXX executable tests\test_resolve_imports.exe
[135/168 1.5/sec] Linking CXX executable tests\test_clone.exe
[136/168 1.5/sec] Linking CXX executable tests\test_isolated_generator.exe
[137/168 1.5/sec] Linking CXX executable tests\test_generator.exe
[138/168 1.5/sec] Linking CXX executable tests\test_coverage.exe
[139/168 1.5/sec] Linking CXX executable tests\test_api_headers_logger.exe
[140/168 1.5/sec] Linking CXX executable tests\test_component.exe
[141/168 1.5/sec] Linking CXX executable tests\test_connection.exe
[142/168 1.5/sec] Linking CXX executable tests\test_logger.exe
[143/168 1.5/sec] Linking CXX executable tests\test_math.exe
[144/168 1.5/sec] Linking CXX executable tests\test_printer.exe
[145/168 1.5/sec] Linking CXX executable tests\test_reset.exe
[146/168 1.6/sec] Linking CXX executable tests\test_validator.exe
[147/168 1.6/sec] Linking CXX executable tests\test_version.exe
[148/168 1.6/sec] Linking CXX executable tests\test_api_headers_namedentity.exe
[149/168 1.6/sec] Linking CXX executable tests\test_units.exe
[150/168 1.6/sec] Linking CXX executable tests\test_api_headers_component.exe
[151/168 1.6/sec] Linking CXX executable tests\test_variable.exe
[152/168 1.6/sec] Linking CXX executable tests\test_api_headers_componententity.exe
[153/168 1.6/sec] Linking CXX executable tests\test_model.exe
[154/168 1.6/sec] Linking CXX executable tests\test_api_headers_entity.exe
[155/168 1.6/sec] Linking CXX executable tests\test_parser.exe
[156/168 1.6/sec] Linking CXX executable tests\test_api_headers_validator.exe
[157/168 1.6/sec] Linking CXX executable tests\test_api_headers_version.exe
[158/168 1.6/sec] Linking CXX executable tests\test_api_headers_variable.exe
[159/168 1.6/sec] Linking CXX executable tests\test_api_headers_generatorprofile.exe
[160/168 1.7/sec] Linking CXX executable tests\test_api_headers_reset.exe
[161/168 1.7/sec] Linking CXX executable tests\test_api_headers_importedentity.exe
[162/168 1.7/sec] Linking CXX executable tests\test_api_headers_model.exe
[163/168 1.7/sec] Linking CXX executable tests\test_api_headers_issue.exe
[164/168 1.7/sec] Linking CXX executable tests\test_api_headers_importsource.exe
[165/168 1.7/sec] Linking CXX executable tests\test_api_headers_parser.exe
[166/168 1.7/sec] Linking CXX executable tests\test_api_headers_printer.exe
[167/168 1.7/sec] Linking CXX executable tests\test_api_headers_units.exe
[168/168 1.7/sec] Linking CXX executable tests\test_api_headers_types.exe
13:38:31: The process "C:\Program Files\CMake\bin\cmake.exe" exited normally.
13:38:31: Elapsed time: 01:47.
```

Anyway, I am going to see whether we can tell CMake to use `/W4` by default and not `/W3` as it does for me using Qt Creator (but not if I use CMake GUI on Windows and then VS 2017...!?).

Reply:
Put `/W4` back in, which means that I am now getting a lot of warnings that read:
```
cl : Command line warning D9025 : overriding '/W3' with '/W4'
```
when building libCellML using Qt Creator on Windows. Ugly!

We use `target_compile_options()` to set compile options, so no idea how to remove `/W3` (and thus prevent those warnings). @hsorby, any ideas?